viewing_party/party.py

```diff
@@ -1,23 +1,352 @@
+    The values of these key-value pairs should be appropriate values
+    If `title` is falsy, `genre` is falsy, or `rating` is falsy, 
+    this function should return `None`
+    
+    '''
+    if title and genre and rating:
+        dict = {"title": title,
+                "genre": genre,
+                "rating": rating}
+        return dict
+    return None
+
+
+def add_to_watched(user_data, movie):
+    '''
+    take two parameters: `user_data`, `movie`
+    the value of `user_data` will be a dictionary with a key `"watched"`, 
+    and a value which is a list of dictionaries representing the movies the user has watched
+    An empty list represents that the user has no movies in their watched list
+    the value of `movie` will be a dictionary in this format:
+    - ```python
+    {
+        "title": "Title A",
+        "genre": "Horror",
+        "rating": 3.5
+    }
+    ```
+    add the `movie` to the `"watched"` list inside of `user_data`
+    return the `user_data`
+    
+    '''
+    #user_data = { “watched”: []}, never watched any movie
+    #user_data = {“watched”: [{“title”: “title A”, “genre”: “horror”, “rating”: 4}]}
+    #movie is a dict, user_data is a dict, watch value is list
+    user_data["watched"].append(movie)
+    return user_data
+    
+
+def add_to_watchlist(user_data, movie):
+    '''
+    take two parameters: `user_data`, `movie`
+    the value of `user_data` will be a dictionary with a key `"watchlist"`, and a value which is a list of dictionaries representing the movies the user wants to watch
+    An empty list represents that the user has no movies in their watchlist
+    the value of `movie` will be a dictionary in this format:
+    ```python
+    {
+        "title": "Title A",
+        "genre": "Horror",
+        "rating": 3.5
+    }
+    ```
+    add the `movie` to the `"watchlist"` list inside of `user_data`
+    return the `user_data`
+    
+    '''
+    user_data["watchlist"].append(movie)
+    return user_data 
+
+
+def watch_movie(user_data, title):
+    '''
+    take two parameters: `user_data`, `title`
+    the value of `user_data` will be a dictionary with a `"watchlist"`and a `"watched"`
+    This represents that the user has a watchlist and a list of watched movies
+    the value of `title` will be a string
+    This represents the title of the movie the user has watched
+    If the title is in a movie in the user's watchlist:
+    remove that movie from the watchlist
+    add that movie to watched
+    return the `user_data`
+    If the title is not a movie in the user's watchlist:
+    return the `user_data`
+    Note: For Waves 2, 3, 4, and 5, your implementation of each 
+    of the functions should not modify `user_data`.
+'''
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)
+            return user_data

```

Comment:
We add the return here to avoid iterating and modifying data at the same time, in this case, you could create one for loop with an if conditional that removes the movie from the watchlist and adds the movie to be watched in one loop. As soon as we make a change we want to stop iterating. Great job!